search-app/src/cmr/search/services/humanizers/humanizer_report_service.clj

```diff
@@ -70,18 +75,25 @@
   (map #(rfm->umm-collection context %) rfms))
 
 (defn- wait-and-retry-for-collection-cache
-  "Wait configurable number of seconds before retrying to get all the collections
-   from collection cache. If collections are still not present, throw warning."
+  "Wait configurable number of milli seconds before retrying configurable number of times
+   to get all the collections from collection cache. If collections are still not present, throw warning."
   [context]
   (info (format "Collection cache is not populated after %d seconds of delay" 
                 (humanizer-report-generator-job-delay)))
-  (Thread/sleep (humanizer-report-generator-job-wait)) 
+  ;; sleep, retry 19 times, then sleep and retry one last time.
+  (Thread/sleep (humanizer-report-generator-job-wait))
+  (let [a (atom (- (humanizer-report-generator-job-retry-number) 1))]
+    (while (and (pos? @a) (not (seq (metadata-cache/all-cached-revision-format-maps context))))
+      (info (format "Humanizer report generator job is sleeping for %d milli-seconds before retry"
```

Comment:
Might want to say something about sleeping before retrying populating the collection cache again. It could be a bit confusing for someone looking in the logs if they don't know we're waiting on the collection cache, not the humanizer report.